Dark Mode for the status site

brandong512 opened the pull request:
- [X] Cookies Implemented
- [X] Sass theming system built
- [X] Toggle button at the top (day/night) according to spec

@akshay1502 
@rohan09-raj 
@whyDontI 

![dark mode for status site](https://user-images.githubusercontent.com/5855338/153381153-3140e6a7-6614-418c-bbe9-c164236bdeec.gif)

Not sure why the long chain of commits, but I'm open to learning more about it/correcting it

rohan09-raj reviewed (commented):
@brandong512 Please check why the tests/checks are failing by clicking on the `details`
![image](https://user-images.githubusercontent.com/78433013/154048500-c1bfc3f5-8277-42eb-98ae-e534ee2935bf.png)

shubham-y commented on src/pages/index.tsx:
```diff
@@ -114,6 +124,8 @@ const Index: FC = () => {
   }, [isLoading, response]);
 
   useEffect(() => {
+    console.log("in use effect: ", checkThemeHistory(document.cookie, query))
```
Please avoid adding console statements to production

brandong512 replied:
No problem I'll go ahead and remove that

shubham-y reviewed (changes requested):
Good work 😃
Requesting some changes

brandong512 commented on src/components/PullRequestList/index.tsx:
```diff
@@ -95,9 +100,14 @@ const PullRequestList: FC<PullRequestListProps> = ({ prType }) => {
   }, []);
+  const [mainDarkMode, setMainDarkMode] = useState(getDefaultOrTransferDark(query))
 
+  const themeSetter = () => {
+    document.cookie = setCookie(!mainDarkMode);
+    setMainDarkMode(!mainDarkMode);
+  }
```
I was actually struggling to do something like this. The issue here is that `document.cookie` is a global argument that relies on the component itself loading. If I try to put this argument in a helper function and it's called immediately before the DOM loads, then an error will show saying `document doesn't exist`.

Any suggestions?

shubham-y commented:
@brandong512 Please add the dark mode feature changes for [Availability panel ](https://status.realdevsquad.com/availability-panel) also which is currently behind a feature flag.

shubham-y reviewed (commented):
The background color of toastify modal does not change after switching from dark mode to light mode
![image](https://user-images.githubusercontent.com/46373689/157806832-2f8473af-8c4d-41a7-b5d1-eb8be8b35930.png)

shubham-y reviewed (commented):
Suggestion: Add a calendar icon with a white border for dark mode
![image](https://user-images.githubusercontent.com/46373689/157807341-d44d15b4-1554-4ec2-8fd1-19fa0acf171a.png)

brandong512 commented:
Sounds good I'll go ahead and start working on these changes. Sorry for the delay!

brandong512 commented:
- [X] added dark mode to availability panel
- [X] added dark mode suggestion for calendar icon 
- [X] fixed dark mode for toast

@akshay1502
@rohan09-raj
@whyDontI
